Topics/Development_Process/Azure_webapp_deployment_with_nodejs.md

```diff
@@ -1,10 +1,191 @@
+
+### Azure App Service
+
 Azure application service allows you to build and host web applications, mobile backends and RESTful APL in the selected programming language without managing the infrastructure. It provides automatic scaling and high availability, supports Windows and Linux, and supports automatic deployment from GitHub, Azure Devops or any Git repository. You can refer to our quick start, tutorials and examples to learn how to use Azure application services.
 
-## Depoly Web app on Azure
+#### Create Web app on Azure
+
+From Microsoft's official documentation, here is a quick tutorial to get an example Node.js project deployed:
 
 [deploy web app with node js on Azure](https://learn.microsoft.com/en-us/azure/app-service/quickstart-nodejs?tabs=linux&pivots=development-environment-vscode)
+
+Alternatively, and possibly easier, you can make a very basic web app with Node.js and Express by doing the following:
+
+Note: Learn more about Node.js and Express apps [here](../Tech_Stacks/Express.md).
+
+1. Create a project directory for the Node.js web app
+2. In the root of your project directory, run `npm init` in the terminal (you should have npm and Node.js installed).
+3. In the terminal, run `npm install express`.
+4. Write the following in the directory in a new file, `server.js` (or whatever name you entered as the entry point during `npm init`).
+    ```javascript
+        const express = require("express");
+
+        const app = express();
+
+        const PORT = 3000
+
+        app.get("/", (req, res) => {
+            res.send(`
+            <h1>CSC301 A2 Demo successfully deployed!</h1>
+            <h2>You did it!</h2>
+            `)
+        });
+
+        // Set up server to listen on port 3000
+        app.listen(PORT, () => {
+            console.log(`Listening on port ${PORT}`);
+        });
+    ```
+
+Note that the file structure inside your project directory should currently look something like this:
+
+```
+├── node_modules
+├── package.json
+├── package-lock.json 
+├── server.js 
+└── .gitignore
+```
+
+Have this code in a Github repository to later integrate with Github actions.
+
+Now, you have two options:
+
+1. Deploy your Web App with the Azure portal (if you proceed with this option, the Azure portal will immediately integrate CI/CD with Github Actions)
+
+    On the Azure Portal, navigate to **App Services**: 
+
+    - Click on the **Create** dropdown and select **Web App**. 
+    - For this simple example, you'll only need to fill out the first tab, **Basics**.
+    - Choose a subscription (Azure for Students if you've used the Github Student Developer Pack), a resource group (or create one), and create a name for your Web App.
+    - Choose `Code` for `Publish`, and the appropriate version of Node for your runtime stack.
+    - The default region and Linux plan defaults should suffice, but make sure to **change the Pricing Plan to Free F1 (Shared infrastructure)**.
```

Comment:
What other options are available? This is good for basic app but maybe point to resources about other options for these plans

Reply:
Good point! While the examples shown in this article are clearly not for production, it's helpful to emphasize this and guide the reader down the right path if this is not the case for them. I've included a disclaimer about this, as well as a link to do more research about the right plan for them.